spec/views/components/accordion.html.erb_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "rails_helper"
+include ComponentTestHelper
+
+def component_name
+  "accordion"
+end
+
+describe "accordion component" do
+  let(:simple_accordion) do
+    [
+      {
+        title: "General information and guidance",
+        panel: "1st panel contents",
+      },
+      {
+        title: "Alternative provision censuses",
+        panel: "2nd panel contents",
+      },
+    ]
+  end
+
+  let(:section1) { ".govuk-accordion__section:nth-child(1)" }
+  let(:section2) { ".govuk-accordion__section:nth-child(2)" }
+
+  it "renders nothing without passed content" do
+    render_component({})
+    expect(rendered).to be_empty
+  end
+
+  it "renders an accordion correctly" do
+    # FIXME: On the original minitest specs there is a test for the value of the id.
+    # It is passing but is a false positive.
+    # There is a bug somewhere as the id is being set as default:
+    # https://github.com/alphagov/govuk_publishing_components/blob/e6512895b33a55e5c5f410baa82658caa66e2622/app/views/govuk_publishing_components/components/_accordion.html.erb#L5
+
+    render_component(sections: simple_accordion)
+    expect(rendered).to have_selector(".govuk-accordion__section")
+
+    # expect(rendered).to have_selector(section1, id: "general-information-and-guidance")
+    expect(rendered).to have_selector(section1 + " .govuk-accordion__section-heading", text: "General information and guidance")
+    expect(rendered).to have_selector(section1 + " .govuk-accordion__section-content", text: "1st panel contents")
+
+    # expect(rendered).to have_selector(section2, id: "alternative-provision-censuses")
+    expect(rendered).to have_selector(section2 + " .govuk-accordion__section-heading", text: "Alternative provision censuses")
+    expect(rendered).to have_selector(section2 + " .govuk-accordion__section-content", text: "2nd panel contents")
+  end
+
+  it "renders accordion with description in a header" do
+    descriptions_accordion = simple_accordion
+    descriptions_accordion[0][:description] = "1st section description"
+
+    render_component(sections: descriptions_accordion)
+
+    assert_select ".govuk-accordion__section-header:nth-child(1) .govuk-accordion__section-summary", text: "1st section description"
+    assert_select ".govuk-accordion__section-header:nth-child(2) .govuk-accordion__section-summary", false
+  end
+
+  xit "renders an accordion with custom section ids" do
+    # FIXME: On the original minitest specs this test is passing but is a false positive.
```

Comment:
Good spot. This seems to be because the original `assert_select` doesn't support an `id` param, so we were never checking for the ID. The [component](https://github.com/alphagov/govuk_publishing_components/blob/master/app/views/govuk_publishing_components/components/_accordion.html.erb) doesn't seem to support this, so arguably we could just get rid of this test.

Reply:
I checked in with @alex-ju 
- The custom ID is not being passed through, so we can remove the test.
- It looks like we can replace this custom component with the regular govuk-publishing-component gem instead. He will pick that work up. Once the partial is removed, we will remove this test.